### ISSUE TITLE
Several issues

### ISSUE THREAD

**Issue by dinaabouzeid** [gangweix/BANet#12]
1. **Why can't reproduce your epe and d3?**
I'm using sceneflow, it loads correctly:

`python train_stereo.py
2026-02-22 00:10:15,702 INFO     [helpers.py:188] Loading pretrained weights from url (https://github.com/rwightman/pytorch-image-models/releases/download/v0.1-weights/mobilenetv2_100_ra-b33bc2c4.pth)
Parameter Count: 5456617
2026-02-22 00:10:16,758 INFO     [stereo_datasets.py:159] Added 22390 from FlyingThings frames_finalpass
2026-02-22 00:10:16,869 INFO     [stereo_datasets.py:173] Added 8664 from Monkaa frames_finalpass
2026-02-22 00:10:16,881 INFO     [stereo_datasets.py:188] Added 4400 from Driving frames_finalpass
2026-02-22 00:10:16,881 INFO     [stereo_datasets.py:382] Adding 35454 samples from SceneFlow
2026-02-22 00:10:16,881 INFO     [stereo_datasets.py:482] Training with 35454 image pairs`

I haven't changed anything in the code except setting:
`os.environ['CUDA_VISIBLE_DEVICES'] = '0'`
instead of 
`os.environ['CUDA_VISIBLE_DEVICES'] = '0,1'`

in both train_stereo.py and evaluate_stereo.py
just because for some reason DataParallel gets stuck before even running.
you can notice this behaviour also with different numbers of gpus, because of batch size and sceneflow's images, it will get stuck either before the first epoch or at the last batch, we can avoid this by setting drop_last=True in:
`data.DataLoader(val_dataset, batch_size=8, 
        pin_memory=True, shuffle=False, num_workers=8)`
but then we'll get different result for the epe as well.

evaluation:
`python evaluate_stereo.py --restore_ckpt checkpoints/sceneflow/banet.pth
2026-02-26 12:51:05,404 INFO     [evaluate_stereo.py:229] Loading checkpoint...
2026-02-26 12:51:05,568 INFO     [evaluate_stereo.py:232] Done loading checkpoint
The model has 5.46M learnable parameters.
2026-02-26 12:51:05,660 INFO     [stereo_datasets.py:159] Added 4370 from FlyingThings frames_finalpass
100%|███████████████████████████████████████████████████████████████████████████████████████████████| 547/547 [01:58<00:00,  4.63it/s]
Validation Scene Flow: 0.691282, 3.153711`

my pip list:

`Package                  Version
------------------------ ---------
absl-py                  2.4.0
annotated-doc            0.0.4
anyio                    4.12.1
certifi                  2026.1.4
click                    8.3.1
cuda-bindings            12.9.4
cuda-pathfinder          1.3.4
filelock                 3.24.3
fsspec                   2026.2.0
grpcio                   1.78.1
h11                      0.16.0
hf-xet                   1.2.0
httpcore                 1.0.9
httpx                    0.28.1
huggingface_hub          1.4.1
idna                     3.11
ImageIO                  2.37.2
Jinja2                   3.1.6
lazy_loader              0.4
Markdown                 3.10.2
markdown-it-py           4.0.0
MarkupSafe               3.0.3
mdurl                    0.1.2
mpmath                   1.3.0
networkx                 3.6.1
numpy                    2.4.2
nvidia-cublas-cu12       12.8.4.1
nvidia-cuda-cupti-cu12   12.8.90
nvidia-cuda-nvrtc-cu12   12.8.93
nvidia-cuda-runtime-cu12 12.8.90
nvidia-cudnn-cu12        9.10.2.21
nvidia-cufft-cu12        11.3.3.83
nvidia-cufile-cu12       1.13.1.3
nvidia-curand-cu12       10.3.9.90
nvidia-cusolver-cu12     11.7.3.90
nvidia-cusparse-cu12     12.5.8.93
nvidia-cusparselt-cu12   0.7.1
nvidia-nccl-cu12         2.27.5
nvidia-nvjitlink-cu12    12.8.93
nvidia-nvshmem-cu12      3.4.5
nvidia-nvtx-cu12         12.8.90
opencv-python            4.13.0.92
packaging                26.0
pillow                   12.1.1
pip                      26.0.1
protobuf                 6.33.5
Pygments                 2.19.2
PyYAML                   6.0.3
rich                     14.3.3
safetensors              0.7.0
scikit-image             0.26.0
scipy                    1.17.0
setuptools               82.0.0
shellingham              1.5.4
sympy                    1.14.0
tensorboard              2.20.0
tensorboard-data-server  0.7.2
tifffile                 2026.2.20
timm                     0.5.4
torch                    2.10.0
torchvision              0.25.0
tqdm                     4.67.3
triton                   3.6.0
typer                    0.24.1
typer-slim               0.24.0
typing_extensions        4.15.0
Werkzeug                 3.1.6`

2. I downloaded your pretrained sceneflow model, why if i used one gpu in evaluation i get the mentioned epe in the paper aka:
0.576362, 2.583245
and if i used more gpus it either get stuck or show wrong values? did you evaluate your model with two gpus?

3. Is there a reason why you commented spatial augmentation in here?
https://github.com/gangweix/BANet/blob/7cb83f7cca99a249e1fcec8ff556e4ac73facd70/banet-2d/core/utils/augmentor.py#L116-L136
